test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -887,3 +887,13 @@ TEST_VM(os, is_first_C_frame) {
   EXPECT_FALSE(os::is_first_C_frame(&cur_frame));
 #endif // _WIN32
 }
+
+TEST_VM(os, open_O_CLOEXEC) {
+#if defined(__APPLE__)
```

Comment:
Does this not work on other OS ie Linux?

Reply:
It works on Linux too and originally I had it for both Mac and Linux, but I figured that since this issue (dropping `FD_CLOEXEC` and relaying on only `O_CLOEXEC`) is focused just on MacOS, then that's the platform we should test.

I can add Linux here though, if you think it would be useful.

Reply:
This should work for all Unices. Can you test it for all !windows platforms?

Also, thanks for adding the test. The reason I did ask for it was that we have that insanely complicated `test/hotspot/jtreg/runtime/8176717/TestInheritFD.java` in the jtreg suite that checks whether UL log files are inherited to child processes (so, not marked with CLOEXEC). I always wondered whether we could replace that one with just a very simple gtest, that tests that UL opens its files with CLOEXEC.